$NetBSD: patch-configure,v 1.2 2024/10/09 04:29:58 ryoon Exp $

Sun audio support.

Enable PIC on NetBSD, even on i386, to avoid text relocations.

Fix install_name_dir on Darwin.

Do not use 'rsync'.

--- configure.orig	2024-09-29 23:31:47.000000000 +0000
+++ configure
@@ -2299,6 +2299,7 @@ HEADERS_LIST="
     sys_resource_h
     sys_select_h
     sys_soundcard_h
+    sys_audioio_h
     sys_time_h
     sys_un_h
     sys_videoio_h
@@ -2447,7 +2448,6 @@ TOOLCHAIN_FEATURES="
     inline_asm_labels
     inline_asm_nonlocal_labels
     pragma_deprecated
-    rsync_contimeout
     symver_asm_label
     symver_gnu_asm
     vfp_args
@@ -3739,6 +3739,8 @@ opengl_outdev_deps="opengl"
 opengl_outdev_suggest="sdl2"
 oss_indev_deps_any="sys_soundcard_h"
 oss_outdev_deps_any="sys_soundcard_h"
+sunau_indev_deps_any="sys_audioio_h"
+sunau_outdev_deps_any="sys_audioio_h"
 pulse_indev_deps="libpulse"
 pulse_outdev_deps="libpulse"
 sdl2_outdev_deps="sdl2"
@@ -5754,6 +5756,7 @@ case $target_os in
     netbsd)
         disable symver
         enable section_data_rel_ro
+        enable pic
         oss_indev_extralibs="-lossaudio"
         oss_outdev_extralibs="-lossaudio"
         enabled gcc || check_ldflags -Wl,-zmuldefs
@@ -5780,7 +5783,7 @@ case $target_os in
         ;;
     darwin)
         enabled ppc && add_asflags -force_cpusubtype_ALL
-        install_name_dir_default='$(SHLIBDIR)'
+        install_name_dir_default=$shlibdir
         SHFLAGS='-dynamiclib -Wl,-single_module -Wl,-install_name,$(INSTALL_NAME_DIR)/$(SLIBNAME_WITH_MAJOR),-current_version,$(LIBVERSION),-compatibility_version,$(LIBMAJOR)'
         enabled x86_32 && append SHFLAGS -Wl,-read_only_relocs,suppress
         strip="${strip} -x"
@@ -6637,6 +6640,7 @@ check_headers malloc.h
 check_headers mftransform.h
 check_headers net/udplite.h
 check_headers poll.h
+check_headers sys/audioio.h
 check_headers pthread_np.h
 check_headers sys/param.h
 check_headers sys/resource.h
@@ -7175,7 +7179,6 @@ enabled makeinfo \
 disabled makeinfo_html && texi2html --help 2> /dev/null | grep -q 'init-file' && enable texi2html || disable texi2html
 perl -v            > /dev/null 2>&1 && enable perl      || disable perl
 pod2man --help     > /dev/null 2>&1 && enable pod2man   || disable pod2man
-rsync --help 2> /dev/null | grep -q 'contimeout' && enable rsync_contimeout || disable rsync_contimeout
 xmllint --version  > /dev/null 2>&1 && enable xmllint   || disable xmllint
 
 check_headers linux/fb.h
